py-polars/tests/unit/test_lazy.py

```diff
@@ -1313,7 +1313,7 @@ def test_quadratic_behavior_4736() -> None:
     ldf.select(reduce(add, (pl.col(fld) for fld in ldf.columns)))
 
 
-@pytest.mark.parametrize("input_dtype", [pl.Utf8, pl.Int64, pl.Float64])
```

Comment:
This pl.utf8 is to test that the epoch string behavior works, but I think it should not be supported as it is a string not following date/datetime format. ie: `pl.DataFrame({"x1":["1234"]}).with_columns(**{ "x1-date": pl.col("x1").cast(pl.Date)})`

Should we keep the support for epoch string?

Reply:
I added it back.

Reply:
Are we sure this is meant to work? 

To be honest it feels a bit strange to have that `pl.Series([<string>]).cast(pl.Date)` could be interpreted as both a format string, or as the number of days since 1970-01-01

I find it hard to believe that anyone would store a timestamp as a string - if they're trying to cast a string to Date, it's almost certainly a date string (and if not, I think it's OK to let the user cast to `Int32` first):
```python
# this looks odd to me
In [9]: print(pl.Series(['1234']).cast(pl.Date))
shape: (1,)
Series: '' [date]
[
        1973-05-19
]
```

My suggestion would be to remove Utf8 from here, treat this as a bug fix, and not double-traverse the array with `is_parsable_as_number` - @stinodego  thoughts?

Reply:
I completely agree, that's why I removed it in the first place but got no feedback on my answer so I put it back, also there is an open issue where this behavior is reported as a bug, https://github.com/pola-rs/polars/issues/10478

Reply:
nice, thanks - ok maybe let's add that as a test case and check and `closes #10478` to the description?

and could we wait for Stijn's response before marking this as resolved please? thanks 🙏 

finally, sorry for having been slow on this PR, been on holiday recently

Reply:
Apologies, haven't b een able to get back to this one. Agree with what you guys are saying - casting `"1234"` to a Date should fail.

Reply:
No worries, I understand that there is a lot of work here to be done and also life happens. Will make the changes. Thanks.